util/capacity-reservation.py

```diff
@@ -58,20 +62,57 @@ def parse_args():
     return parser.parse_args()
 
 
-def create_reservation(client, instance_type, zone, count, tag_path, dry_run):
+def _retrieve_tags(client, instance_id):
+    """Retrieve tags from an EC2 instance"""
+    response = client.describe_tags(
+        Filters=[
+            {
+                'Name': 'resource-id',
+                'Values': [instance_id]
+            }
+        ]
+    )
+    if 'Tags' in response:
+        tags = [
+            {'Value': tag['Value'], 'Key': tag['Key']}
+            for tag in response['Tags']
+            if tag['Key'] in ['Department', 'Name', 'Project', 'SubProject']
+            ]
+        return tags
+    else:
+        metadata = response['ResponseMetadata']
+        print(f'Problem retrieving tags: {metadata}')
+        sys.exit(1)
+
+
+def _resolve_tags(client, args):
+    """If there's a tag file path in arguments, load it, otherwise,
```

Comment:
What is the use case for the `tag file`?  is that to allow you to manually override the instances with a custom set of tags?

Reply:
Could be used to override, allows one to avoid looking up an instance to get the tags all the time

Reply:
yes, it allows the override, or a way around continuously having to look up an instance id